pkg/statistics/handle/syncload/stats_syncload.go

```diff
@@ -232,18 +232,34 @@ func (s *statsSyncLoad) HandleOneTask(sctx sessionctx.Context, lastTask *statsty
 		if result.Err == nil {
 			slr := result.Val.(*stmtctx.StatsLoadResult)
 			if slr.Error != nil {
+				if ok := updateNeededItemTaskRetryCountAndCheck(task); !ok {
```

Comment:
Maybe we can try `!updateNeededItemTaskRetryCountAndCheck`.

Reply:
I has been refactored.